great_expectations/core/batch_definition.py

```diff
@@ -87,7 +87,7 @@ def identifier_bundle(self) -> _EncodedValidationData:
         )
         batch_definition_bundle = _IdentifierBundle(
             name=self.name,
-            id=self.id,
+            id=str(self.id) if self.id else None,
```

Comment:
Ugh, looks like for everything else we use `str` or `str | UUID` (but must always be str) for an `id`.

Reply:
Ugh yeah we could update the identifier bundle to use UUIDs but this is the path of least resistance